code-challenges.test.js

```diff
@@ -57,7 +107,27 @@ var numbersToAdd2 = [0, 7, -8, 12]
 
 var numbersToAdd3 = []
 // Expected output: []
-
+test("returns the array that composes a number of the fibonacci sequence. ",()=>{
+    expect(summation(numbersToAdd1).toEqual([2, 6, 51, 60]))
+    expect(summation(numbersToAdd2).toEqual([0, 7, -8, 12]))
+    expect(summation(numbersToAdd3).toEqual([]))
+})
 
 
 // b) Create the function that makes the test pass.
+//create and array that adds each number to its next index.
+// this can be done using the fibonnaci sequence and taking in an array and iterating through the array while taking the previous index of each array.
+// using the 1st index setting it to 0 then updating it from there (however here is also my problem i dont understand how to have it update.)
+const summation = (array) => {
+    for (let index = 0; index < array.length; index++) {
+        let x = array[index]           // sets x equal to the second place x  0 + 1 would make the x2 eqaul to 1
+        let x2 = 0                      // placement of this takes care of the initial undefined issue
+        let sum = x + x2                // this adds together each number in the array
+        x2 = array[index - 1]           // this then sets the index equal 
+
+        console.log ("1st",array[index])
+        console.log ('2nd',array[index - 1])
+        console.log ("sum",sum)
+        
+    }
+}
```

Comment:
The beauty here is that we always want to return an array of equal length. the .map() method does this for us nicely and we can use an accumulator to hold our summed number and set the values as we like!